src/Nethermind/Nethermind.Init/Steps/InitializeBlockchain.cs

```diff
@@ -140,6 +147,11 @@ private Task InitBlockchain()
                     No.Pruning,
                     Persist.EveryBlock,
                     getApi.LogManager);
+
+                worldState = setApi.WorldState = new WorldState(
```

Comment:
Why not enable if pruning is disabled?

Reply:
This mostly targets fixing issues while pruning. Not sure if we want to target any issues during archive sync, WDYT?

Reply:
I'd turn it on all the time, who knows what is causing trie exception. Also, I'd add a flag to turn it off also.